packages/celotool/src/lib/k8s-oracle/base.ts

```diff
@@ -0,0 +1,110 @@
+import { getFornoUrl, getFornoWebSocketUrl, getFullNodeHttpRpcInternalUrl, getFullNodeWebSocketRpcInternalUrl } from 'src/lib/endpoints'
+import { envVar, fetchEnv, fetchEnvOrFallback } from 'src/lib/env-utils'
+import { installGenericHelmChart, removeGenericHelmChart, upgradeGenericHelmChart } from 'src/lib/helm_deploy'
+
+const helmChartPath = '../helm-charts/oracle'
+
+/**
+ * Represents the identity of a single oracle
+ */
+export interface OracleIdentity {
+  address: string
+}
+
+export interface BaseOracleDeploymentConfig {
+  context: string
+  identities: OracleIdentity[]
+  useForno: boolean
+}
+
+export abstract class BaseOracleDeployer {
+  protected _deploymentConfig: BaseOracleDeploymentConfig
+  private _celoEnv: string
+
+  constructor(deploymentConfig: BaseOracleDeploymentConfig, celoEnv: string) {
+    this._deploymentConfig = deploymentConfig
+    this._celoEnv = celoEnv
+  }
+
+  async installChart() {
+    return installGenericHelmChart(
+      this.celoEnv,
+      this.releaseName,
+      helmChartPath,
+      await this.helmParameters()
+    )
+  }
+
+  async upgradeChart() {
+    return upgradeGenericHelmChart(
+      this.celoEnv,
+      this.releaseName,
+      helmChartPath,
+      await this.helmParameters()
+    )
+  }
+
+  async removeChart() {
+    await removeGenericHelmChart(this.releaseName)
+  }
+
+  async helmParameters() {
+    const httpRpcProviderUrl = this.deploymentConfig.useForno
+      ? getFornoUrl(this.celoEnv)
+      : getFullNodeHttpRpcInternalUrl(this.celoEnv)
+    const wsRpcProviderUrl = this.deploymentConfig.useForno
+      ? getFornoWebSocketUrl(this.celoEnv)
+      : getFullNodeWebSocketRpcInternalUrl(this.celoEnv)
+    return [
+      `--set environment.name=${this.celoEnv}`,
+      `--set image.repository=${fetchEnv(envVar.ORACLE_DOCKER_IMAGE_REPOSITORY)}`,
+      `--set image.tag=${fetchEnv(envVar.ORACLE_DOCKER_IMAGE_TAG)}`,
+      // TODO change this & the corresponding oracle env var
```

Comment:
Still doing this, waiting on my change in the oracle client to make the init retry logic not just for azure hsms. Will clean up soon